### PR TITLE
Bug 1694878: Always honor OAuth tokens

### PR DESCRIPTION
**Pull request by enj**
Regardless of external configuration, OAuth tokens should always work.

This prevents issues where multiple masters exist but only some will honor OAuth tokens due to differences in configuration.

Bug 1694878

Signed-off-by: Monis Khan <mkhan@redhat.com>

### CODE REVIEW

File: pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch_authenticator.go [hunk 1]
```diff
@@ -120,18 +120,20 @@ func newAuthenticator(
...
+	// we assume it supports the bootstrap oauth user by setting a non-nil session config
+	if oauthConfig == nil {
+		oauthConfig = &osinv1.OAuthConfig{
+			SessionConfig: &osinv1.SessionConfig{},
+		}
+	}
+
 	// OAuth token
 	// this looks weird because it no longer belongs here (needs to be a remote token auth backed by osin)
 	if oauthConfig != nil || len(authConfig.OAuthMetadataFile) > 0 {
```
Comment (stlaz) [inline pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch_authenticator.go]: No need to do `oauthConfig != nil` anymore.

File: pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch_authenticator.go [hunk 2]
```diff
@@ -120,18 +120,20 @@ func newAuthenticator(
...
-		// if we have no OAuthConfig but have an OAuthMetadataFile, we still need to honor OAuth tokens
-		// to keep the checks below simple, we build an empty OAuthConfig
-		// since we do not know anything about the remote OAuth server's config,
-		// we assume it supports the bootstrap oauth user by setting a non-nil session config
-		if oauthConfig == nil {
```
Comment (stlaz) [inline pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch_authenticator.go]: :cry:

**Comment by deads2k** [comment]
this is the correct place to fix it.  What about something like https://github.com/openshift/origin/pull/22687/files that removes the entire conditional instead?

**Pull request merged**